### PR TITLE
Component/tooltip

### PR DESCRIPTION
**Pull request by haild173094**

### CODE REVIEW

File: docs/components/Tooltip/README.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+To continue using Shopify, this amount must be paid immediately.
+
+</DoDont>
+
+---
+
+## Related components
+
+- To make helpful content more visible to merchants, use the help text portions of form components such as [text fields](https://polaris.shopify.com/components/selection-and-input/text-field), [footer help](https://polaris.shopify.com/components/navigation/footer-help), or [an inline link to help](https://polaris.shopify.com/components/link)
```
Comment (juzser) [inline docs/components/Tooltip/README.md:83]: Update to our component links

File: src/components/EventListener/EventListener.vue [hunk 2]
```diff
@@ -0,0 +1,57 @@
+<template>
```
Comment (juzser) [inline src/components/EventListener/EventListener.vue:1]: Already have this component. 
Or consider to use `use-event-listener` instead

File: src/components/EventListener/index.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+export { default as EventListener } from './EventListener.vue';
```
Comment (juzser) [inline src/components/EventListener/index.ts:1]: .

File: src/components/Popover/components/Pane/Pane.vue [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  /** Default slot */
+  default: (_: VueNode) => null;
+}
+
+type Emits = {
+  /** Called when scrolled to the bottom of the scroll area */
+  'scrolled-to-bottom': [];
+}
+
+const styles = useCssModule();
```
Comment (juzser) [inline src/components/Popover/components/Pane/Pane.vue:56]: . use import instead

File: src/components/Popover/components/Pane/Pane.vue [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  ? {
+    height: props.height,
+    maxHeight: props.height,
+    mingHeight: props.height,
+  }
+  : undefined);
+</script>
+
+<style lang="scss" module>
+@import '@polaris/components/Popover/Popover.scss';
```
Comment (juzser) [inline src/components/Popover/components/Pane/Pane.vue:91]: .

File: src/utilities/math.ts [hunk 6]
```diff
@@ -0,0 +1,147 @@
+import {Rect} from '../../../utilities/geometry';
```
Comment (juzser) [inline src/utilities/math.ts:1]: Where is this? Use shortcut alias instead

File: src/components/Tooltip/components/TooltipOverlay/TooltipOverlay.vue [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+  styles.TooltipOverlay,
+  overlayDetails.value?.measuring && styles.measuring,
+  !overlayDetails.value?.measuring && styles.measured,
+  props.instant && styles.instant,
+  overlayDetails.value?.positioning === 'above' && styles.positionedAbove,
+  ));
+const contentClassName = computed(() => classNames(
+  styles.Content,
+  props.width && styles[props.width],
+  ));
```
Comment (juzser) [inline src/components/Tooltip/components/TooltipOverlay/TooltipOverlay.vue:71]: tab size

File: src/components/Tooltip/components/TooltipOverlay/TooltipOverlay.vue [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+// Computed
+const overlayDetails = computed(() => overlayRef.value?.overlayDetails);
+const containerClassName = computed(() => classNames(
+  styles.TooltipOverlay,
+  overlayDetails.value?.measuring && styles.measuring,
+  !overlayDetails.value?.measuring && styles.measured,
+  props.instant && styles.instant,
+  overlayDetails.value?.positioning === 'above' && styles.positionedAbove,
+  ));
```
Comment (juzser) [inline src/components/Tooltip/components/TooltipOverlay/TooltipOverlay.vue:67]: tab size

File: src/components/Tooltip/components/TooltipOverlay/TooltipOverlay.vue [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+      fill="none",
+    )
+      TailDownPathsVue(v-if="overlayDetails?.positioning === 'above'")
+      TailUpPaths(v-else)
+    div(
+      :id="id",
+      role="tooltip",
+      :class="contentClassName",
+      :style="{ ...contentStyles, ...style }",
+      :aria-label="accessibilityLabel ? i18n.translate('Polaris.Tooltip.overlayLabel', { label: accessibilityLabel }) : undefined",
```
Comment (juzser) [inline src/components/Tooltip/components/TooltipOverlay/TooltipOverlay.vue:29]: `a && b`

File: src/components/Tooltip/Tooltip.vue [hunk 10]
```diff
@@ -0,0 +1,253 @@
...
+  /** Whether the tooltip's content remains open after clicking the activator */
+  persistOnClick?: boolean;
+}
+
+const props = withDefaults(defineProps<TooltipProps>(), {
+  preferredPosition: 'above',
+  activatorWrapper: 'span',
+  width: 'default',
+  padding: 'default',
+  active: undefined,
```
Comment (juzser) [inline src/components/Tooltip/Tooltip.vue:97]: .

File: src/components/Tooltip/Tooltip.vue [hunk 11]
```diff
@@ -0,0 +1,253 @@
...
+const {
+  value: persist,
+  toggle: togglePersisting
+} = useToggle(Boolean(props.active) && Boolean(props.persistOnClick));
+
+const activatorNode = ref<HTMLElement | null>(null);
+const activatorContainer = ref<HTMLElement | null>(null);
+const mouseEntered = ref(false);
+const shouldAnimate = ref(!props.active);
+const hoverDelayTimeout = ref<NodeJS.Timeout | null>(null);
```
Comment (juzser) [inline src/components/Tooltip/Tooltip.vue:131]: Is the type is `number` instead?

File: src/components/PositionedOverlay/PositionedOverlay.vue [hunk 12]
```diff
@@ -0,0 +1,341 @@
...
+  observer.value.observe(activator, OBSERVER_CONFIG);
+}
+
+defineExpose({
+  forceUpdatePosition,
+  overlayDetails,
+});
+</script>
+
+<style lang="scss" module>
```
Comment (juzser) [inline src/components/PositionedOverlay/PositionedOverlay.vue:339]: .

**Review by juzser — CHANGES_REQUESTED** [review]

File: src/components/Tooltip/Tooltip.vue (hunk 11 above)
Comment (haild173094) [reply src/components/Tooltip/Tooltip.vue]: Screenshot from 2024-01-26 11-35-03
Wrong type when assign value :thinking:

File: docs/components/Tooltip/README.md [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+
+To continue using Shopify, this amount must be paid immediately.
+
+</DoDont>
+
+---
+
+## Related components
+
+- To make helpful content more visible to merchants, use the help text portions of form components such as [text fields](https://polaris.shopify.com/components/TextField), [footer help](https://polaris.shopify.com/components/FooterHelp), or [an inline link to help](https://polaris.shopify.com/components/Link)
```
Comment (juzser) [inline docs/components/Tooltip/README.md:83]: component

**Review by juzser — CHANGES_REQUESTED** [review]

**Review by juzser — APPROVED** [review]

**Pull request merged**